---
title: "Install organisms from NCBI"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    self_contained: false
    fig_caption: true
    highlight: kate
    theme: cerulean
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 5
---


# Introduction

The goal of this tutorial is to explain how to install genomes from NCBI in your RSAT server. 


# Prerequisite

This protocol assumes that you have a working instance of the RSAT server, and that you are familar with the Unix terminal. 

# Protocol

We illustrate the installation procedure with a bacterial genome, since NCBI is the main source of prokaryote genomes for RSAT (<http://prokaryotes.rsat.eu/>). The same procedure can however apply to other taxa, as far as they are available on the NCBI FTP server (<ftp://ftp.ncbi.nlm.nih.gov/genomes/>).

## Data source

Genomes are downloaded from the refseq repository: <ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq>

## Getting the list of available species

The first step to perform with the command `Ã¬nstall-organisms` is to get the list of organisms available for a given taxonomic group. 

The NCBI FTP genome site is organised by "main" **taxonomic groups**: 

- [archaea](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/)
- [bacteria](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/)
- [fungi](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/)
- [invertebrate](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/invertebrate/)
- [plant](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plant/)
- [mitochondrion](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/mitochondrion/)
- [plasmid](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plasmid)
- [plastid](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plastid)
- [protozoa](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/protozoa/)
- [vertebrate_mammalian](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/)
- [vertebrate_other](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_other/)
- [viral](ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/viral/)

The option `-task available` returns the complete list of available species for the specified group. 

The command below gets the list of all available bacteria and stores them in a text file. It can easily be adapted to another group by changing the environment variable `GROUP`. 

```{bash available_bacteria, eval=FALSE}
## Get curret date to use as suffix for output files
export TODAY=`date +%Y-%m-%d`

## Choose a taxonomic group of interest
export GROUP=bacteria

## Get the list of bacterial species available on the NCBI FTP site
install-organism -v 1 -group ${GROUP} -task available \
  -o available_${GROUP}_${TODAY}.txt
  
## Count the number of species.
## Note: grep is used to filter out the comment lines.
grep -v '^;' available_${GROUP}_${TODAY}.txt | wc -l
```

We can of course do the same for other taxa, for example Archaea, Fungi or Mammalian. 

```{bash available_genomes, eval=FALSE}
export NCBI_GROUPS='archaea bacteria fungi invertebrate plant protozoa vertebrate_mammalian vertebrate_other viral'

echo "NCBI_GROUPS ${NCBI_GROUPS}"
for ncbi_group in $NCBI_GROUPS; do \
  echo "Getting list of NCBI species for ${ncbi_group}" ; \
  install-organism -v 1 -group ${ncbi_group} -task available \
    -o available_${ncbi_group}_${TODAY}.txt ; \
done


## Approx count of species per taxa
## (each file starts with 15 comment lines, which are be substracted from each count)
wc -l available_*_${TODAY}.txt |  sort -nr  | awk '{print "|\t"$1-15"\t| "$2"\t|"}' 

```

Note: for some species, several strains have been sequenced. The number of genomes available at NCBI thus exceeds by far the number of distinct species. 


### Number of species per group

Here is the result on Nov 26, 2023

| Number of species | Taxonomic group |
|---------|--------------------------|
|       50033   | available_bacteria_2023-11-26.txt     |
|       14148   | available_viral_2023-11-26.txt        |
|       1176    | available_archaea_2023-11-26.txt      |
|       566     | available_fungi_2023-11-26.txt        |
|       377     | available_invertebrate_2023-11-26.txt |
|       361     | available_vertebrate_other_2023-11-26.txt     |
|       223     | available_vertebrate_mammalian_2023-11-26.txt |
|       177     | available_plant_2023-11-26.txt        |
|       94      | available_protozoa_2023-11-26.txt     |
|       67275   | total |


**Notes** 

- The **number of available genomes** exceeds by far the number of species, because some species are represented by different strains, in particular for bacteria (for some bacterial species of interest for health, several thousands of strains have been sequenced). 

- Some of these genomes are only partly sequenced, or poorly assembled, and the level of annotation is quite variable. NCBI subdivided these available genomes in 4 categories depending on the level of completion of the sequencing project. 


## Installing the genome for a given species

### Listing the available strains for the species of interest

```{bash eval=FALSE}
## Select a species of interest 
## (can be adapted to your needs)
export SPECIES=Escherichia_coli


## Some examples of alternative set of parameters
# export SPECIES=Rhodospirillum_centenum
# export SPECIES=Rhizobium_etli
# export SPECIES=Sinorhizobium_meliloti

## Download the list of strains for the selected species 
## (e.g. Escherichia_coli)  
install-organism -v 1 -group bacteria \
  -species ${SPECIES} \
  -task list

## This list can be stored in a tab-separated text file 
install-organism -v 1 -group bacteria   -species ${SPECIES}   -task download,list -o strain-list_${SPECIES}.tsv

```



### Downloading a selected strain

```{bash eval=FALSE}
## Select a species and straint of interest 
## (can be adapted to your needs)

export SPECIES=Escherichia_coli
export STRAIN=GCF_000005845.2_ASM584v2
export TAXON=Gammaproteobacteria

# export SPECIES=Rhodospirillum_centenum
# export STRAIN=GCF_000016185.1_ASM1618v1
# export TAXON=Rhodospirillum

## An example of alternative set of parameters
# export SPECIES=Rhizobium_etli
# export STRAIN=GCF_000092045.1_ASM9204v1
# export TAXON=Rhizobiales

## Yet another example, just for Mere
# export SPECIES=Sinorhizobium_meliloti
# export STRAIN=GCF_000006965.1_ASM696v1
# export TAXON=Rhizobiales

## Download the genome(s) for a given species 
## (e.g. Escherichia_coli)  
install-organism -v 1 -group bacteria \
  -species ${SPECIES} \
  -strain ${STRAIN} \
  -task download,list
```

**Notes**

1. The option `-species` should be followed by a string concatenating the Genus (with a leading uppercase) and species (in lowercases) separated by an undescore character (`_`), with exactly the same spelling and cases as on the NCBI FTP refseq site. 

2. Downloaded genomes will be installed in a folder `$RSAT/downloads/refseq` on your computer. Each strain is stored in a separate folder, with the same organistion as on the NCBI FTP site. 

```{r list_downloaded_genomes, eval=FALSE}
ls -1 $RSAT/downloads/refseq/bacteria/${SPECIES}/latest_assembly_versions
```


3. **Beware:** for some bacterial species of particular interest (e.g. important human pathogens), severak thousands of strains have been sequenced. It might cost a lot of time and disk space to download all of them. TO avoid that, the option `-strain` enables to specify the full strain name (more precisely the subfolder corresponding to the strain of interest in the species folder on the NCBI FTP site). 


### Parsing a selected genome from NCBI

Parsing consists in extracting the information from a text-formatted file, in order to organise it in a computable structure. 

```{bash eval=FALSE}
## Parse a given strain of a given genome 
## (e.g. strain GCF_000016185.1_ASM1618v1 of Rhodospirillum centenum) 
install-organism -v 2 -group bacteria \
  -species ${SPECIES} \
  -strain ${STRAIN} \
  -task parse
```

**Note:** if the option `-strain` is omitted, `install-organisms` parses all the strains found in the download folder of the selected organism. 


For each parsed genome, the program creates a folder whose name is the concatenation of the genus, species and strain names:  `$RSAT/public_html/data/[Genus_species_strain]`


From now on we will directly refer to this concatenated name as the "Organism name" (generally denoted by the option `-org` in RSAT commands. 

```{bash eval=FALSE}
## Check the parsing result
ls -l ${RSAT}/public_html/data/genomes/${SPECIES}_${STRAIN}/genome/

## Measure the disk space occupied by this genome (note: this will be increased in the subsequent steps)
du -sm ${RSAT}/public_html/data/genomes/${SPECIES}_${STRAIN}/genome/
```

This gives the number of megabases occupied by the genome folder. 

### Adding the parsed genome to the list of supported genomes

In order to make the parsed genome available for RSAT, you need to run the configuration task.

Note that for the subsequent tasks, the options `-species` and `-strains` are not used anymore, we will instead specify "organisms" installed on RSAT, identified by the concatenation of Genus, species and strain. 

```{bash eval=FALSE}
## Configure the organism for RSAT
install-organism -v 1 -org ${SPECIES}_${STRAIN} -source NCBI -task config


## Check that the organism has wel been declared
supported-organisms -taxon ${TAXON} -return ID,source,last_update | grep ${STRAIN}

```

### Finalizing the installation

Several more tasks need to be performed before we can consider this organism as "installed". 

The simplest way to run it is to select the "default" tasks. 

```{bash eval=FALSE}
## Run the installation task (retrieve upstream sequences compute background oligo and dyad frequencies, index fasta genome for bedtols, ...)
install-organism -v 1 -org ${SPECIES}_${STRAIN} -task default

```

#### Detailed list of supported installation tasks (default and others)

| Task | Meaning                                     |
|-------------|-------------------------------------------------|
| default | run all the default tasks useful for the installation of a genome (this is a subset of the tasks specified below) |
| start_stop | compute trinucleotide frequencies in all the start and stop codons (validation of the correspondance between genome sequences and gene coordinates) |
| allup | Retrieve all upstream sequences in order to compute the background models (oligonucleotide and dyad frequencies) |
| seq_len_distrib | Compute the distribution of upstream sequences |
| genome_segments | SPlit the genome into segments corresponding to different genomic region types: genic, intergenic, ... |
| upstream_freq | Compute oligonucleotide or/and dyad frequencies in upstream sequences (must be combined with tasks oligos and/or dyads) |
| genome_freq | Compute oligonucleotide or/and dyad frequencies in the whole genome (must be combined with tasks oligos and/or dyads) |
| protein_freq | Compute amino acid, di- and tri-peptide frequencies in all protein sequences |
| protein_len | Compute distribution of protein lengths |
| oligos | Compute oligonucleotide frequencies (size 1 to 8) in a given type of sequences. Must be combined with stasks upstream or genomic |
| dyads | Idem for dyads, i.e. spaced pairs of trinucleotides |
| fasta_genome | convert genome sequence in fasta in order to make it usable by bedtools | 
| chrom_sizes | compute chromosome sizes from genomic sequences |
| index_bedtools | index genome to enable its ultra-fast processing by bedtools |
| uninstall | uninstall a genome, i.e. stop displaying it as available. Beware, this does not erase the genome folder, for this you can use the option `-task erase`|
| erase | Erase the whole folder of your organism in `$RSAT/public_html/data` |

### Checking that the organism is well supported

```{bash eval=FALSE}
## Get all the organisms the taxon of interest
supported-organisms -taxon ${TAXON}
```

### Checking the correctness of gene annotations by computing start/stop codon frequencies

```{bash eval=FALSE}
## Retrieve all the start codon sequences
retrieve-seq -all -org ${SPECIES}_${STRAIN} -from 0 -to 2 -o start_seq.fasta

## Check the result
head -n 20 start_seq.fasta

## Check that most genes have ATG as start codon
oligo-analysis -v 1 -i start_seq.fasta  -l 3 -1str -return occ,freq -sort | more

```

You can do the similar test with the stop codon frequencies. Note that nothing forces you to store the stop codong sequences in a local file, you can use the pipe symbol ` | ` to concatenate two commands. 

```{bash eval=FALSE}
## Compute all trinucleotide frequencies in the stop codons
retrieve-seq -all -org ${SPECIES}_${STRAIN} \
   -type downstream -from -1 -to -3 -o stop_seq.fasta

## Check the beginning of the result file
head -20 stop_seq.fasta

## Check that most genes have ATG as start codon
oligo-analysis -v 1 -i stop_seq.fasta  -l 3 -1str \
   -return occ,freq -sort | more

```


****
## Selecting organisms with Complete genomes or Chromosomes

For each taxonomic group, NCBI provides a summary file with indications of the sequencing status for each genome project. Here is the link for Bacteria (you can easily adapt it for other groups). 

- <ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt> 

This table contains many columns with valuable information for genome management. The column contain is indicated in a separate file. 

- <ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt>

Genome assemblies are classigied in four levels ([PMCID: PMC4702866](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702866/)).

1. **Contig:** Assemblies that include only contigs.
2. **Scaffold:** Includes both scaffolds and contigs.
3. **Chromosome:** Includes chromosome or linkage groups, plus scaffolds and contigs.
4. **Complete genome:** Assemblies for which all molecules are fully sequenced.

We can use this assembly summary  to select only those genomes with the a completion level that we consider appropriate. 


For example, for the RSAT prokaryote server (<http://prokaryote.rsat.eu>), we restrict the installation to genomes with at least one fully sequenced chromosome (categories "Chromosome" and "Complete Genome").

```{bash eval=FALSE}
## Get current date to use as suffix for output files
export TODAY=`date +%Y-%m-%d`

## Choose a taxonomic group of interest
export GROUP=bacteria

## Set a file name for the assembly file
ASSEMBLY_URL=ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/${GROUP}/assembly_summary.txt
ASSEMBLY=ncbi_organisms_assembly_summary_${GROUP}_refseq_${TODAY}.tsv
COMPLETE=ncbi_organisms_assembly_summary_${GROUP}_complete_${TODAY}.tsv
ONECHROM=ncbi_organisms_assembly_summary_${GROUP}_chromosome_${TODAY}.tsv
ONECHROM_OR_COMPLETE=ncbi_organisms_assembly_summary_${GROUP}_chromosome_or_complete_${TODAY}.tsv

## Dowload the assembly summary from NCBI FTP site
wget -nc ${ASSEMBLY_URL} --output-document ${ASSEMBLY} 

## Count the number of organisms (remove comments first) 
grep -v '^#' ${ASSEMBLY} | wc -l

## Select only the genomes with a Complete or Chromosome annotation in the field assembly_level
awk -F "\t" '$12=="Chromosome" && $11=="latest"{print $0}' \
      ${ASSEMBLY} \
      > ${ONECHROM}
      
awk -F "\t" '$12=="Complete Genome" && $11=="latest"{print $0}' \
      ${ASSEMBLY} \
      > ${COMPLETE}


## Merge summary tables of species with at least one sequenced chromosome
cat ${COMPLETE} ${ONECHROM} >  ${ONECHROM_OR_COMPLETE}

## Count the number of selected species
wc -l ${COMPLETE} ${ONECHROM} ${ONECHROM_OR_COMPLETE} ${ASSEMBLY} 

```

The assembly summary also contains information about the status of the genome for the biologist community, with three possible statuses: *reference genome*, *representative genome*, or *na* (non attributed).

```{bash reference_genomes, eval=FALSE}
REFERENCE=ncbi_organisms_assembly_summary_${GROUP}_reference_${TODAY}.tsv
REPRESENTATIVE=ncbi_organisms_assembly_summary_${GROUP}_representative_${TODAY}.tsv
REPRESENTATIVE_COMPLETE=ncbi_organisms_assembly_summary_${GROUP}_representative_complete_${TODAY}.tsv


## Select reference genomes
awk -F "\t" '$5=="reference genome" && $11=="latest"{print $0}' \
      ${ASSEMBLY} \
      > ${REFERENCE}

## Select representative genomes
awk -F "\t" '$5=="representative genome" && $11=="latest"{print $0}' \
      ${ASSEMBLY} \
      > ${REPRESENTATIVE}

## Select representative and complete genomes
awk -F "\t" '$5=="representative genome" && $12=="Complete Genome"  && $11=="latest"{print $0}' \
      ${ASSEMBLY} \
      > ${REPRESENTATIVE_COMPLETE}

## Print the umbers of reference or representative genomes
wc -l ${REFERENCE} ${REPRESENTATIVE_COMPLETE} ${REPRESENTATIVE} ${COMPLETE} ${ONECHROM} ${ONECHROM_OR_COMPLETE} ${ASSEMBLY} 

```



*****
## Installing many genomes

We can install multiple species in a single command by specifying a file containing a list of species names (one species name per row). 


### Download

**Beware**: 

- This command will download thousands of genomes  on your hard drive, which will undoubtedly occupy some disj space.

- The download can take a considerable time depending on the number of selected species and bandwidth of your computer. 

### Downloading genome files from NCBI

```{bash downloading_many_genomes, eval=FALSE}
## File with the list of species to install
SPECIES_TO_INSTALL=${GROUP}_species_to_install.txt

## Get the two first words of the 8th column as Genus and species names
cut -f 8 ${COMPLETE} \
  | sed 's/ /_/g' \
   > ${SPECIES_TO_INSTALL}

## Count the number of species to instal
wc -l ${SPECIES_TO_INSTALL}

## Download all the selected species
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task download
   

## Check the list of downloaded organisms 
## Note: organism = Genus + species + strain
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task list

```

### Parsing

```{bash parsing_many_genomes, eval=FALSE}

## Parse all organisms
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task list,parse


```

### Parsing genomes from NCBI files

Since the number of genomes to install can be important, we will run `install-organism` with the option `-batch`, in order to parallelise the installation steps via a job scheduler. This job scheduler must have been previously defined in RSAT configuration files. 

```{bash eval=FALSE}
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task parse -batch

make -f ${RSAT}/makefiles/server.mk  watch_jobs
```


### RSAT installation


### Default installation tasks

```{bash installing_many_genomes, eval=FALSE}

## Run default installation tasks
## NOTE: if you need to install hundreds of genomes, this can take a while. ## In such case, it is worth testing the option -batch (see below)
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task config,list,default

```

### Managing installation tasks with a job scheduler

If your RSAT server has been configured adequately, you can add the option `-batch` to the commands above in order to send the installation tasks to a job scheduler. 

```{bash installing_many_geonmes_batch, eval=FALSE}
ORG_TO_INSTALL=${GROUP}_organisms_to_install.txt

## Parse all organisms
install-organism -v 2 -group ${GROUP} \
   -species_file ${SPECIES_TO_INSTALL} -task list,parse \
  -batch


## Get the list of organisms from the list of parsed species
install-organism -v 2 -group ${GROUP} \
  -species_file ${SPECIES_TO_INSTALL} -task list \
  > ${ORG_TO_INSTALL}

## Count the number of species and organisms to install
SPECIES_DO_INSTALL_NB=`grep -v '^;' ${SPECIES_TO_INSTALL} | wc -l  | awk '{print $1}'`
ORG_DO_INSTALL_NB=`grep -v '^;' ${ORG_TO_INSTALL} | wc -l | awk '{print $1}'`
echo "To install: ${ORG_DO_INSTALL_NB} organisms,  ${SPECIES_DO_INSTALL_NB} distinct species"

## First retrieve all upstream sequences (cannot be done in batch mode)
install-organism -v 2  \
   -org_file ${ORG_TO_INSTALL} -task config,allup; 
   
## Send the other installation tasks to the job scheduler
install-organism -v 2 -org_file ${ORG_TO_INSTALL}  -task  start_stop,seq_len_distrib,genome_segments,upstream_freq,genome_freq,protein_freq,protein_len,oligos,dyads,fasta_genome,fasta_genome_rm,chrom_sizes,index_bedtools -batch

```

## Check the installation

TO BE WRITTEN



