---
title: "Installing genomes from EnsemblGenomes"
author: "Bruno Contreras Moreira"
date: '`r Sys.Date()`'
output:
  html_document:
    self_contained: false
    fig_caption: true
    highlight: kate
    theme: cerulean
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 5
---

# Introduction

This document explains how to install genomes and annotations from Ensembl Genomes using mostly the [FTP](ftp://ftp.ensemblgenomes.org/pub) site.
Gene Ontology terms are optional and are obtained from [BioMart](http://plants.ensembl.org/biomart/martview) instead.

Note that while Ensembl covers Vertebrates, Ensembl Genomes (EG, also known as Non Vertebrates, NV) 
includes the other divisions (Protists, Fungi, Plants, Bacteria, Metazoa). 
These instructions have mostly been tested with **Ensembl Plants**.

This protocol uses the Makefile 
[ensemblgenomes_FTP_client.mk](https://github.com/rsa-tools/rsat-code/blob/master/makefiles/ensemblgenomes_FTP_client.mk),
which can also be used to install arbitrary genomes from other sources from FASTA and GTF files,
as explained in 
[Installing organisms from FASTA and GTF files](https://rsa-tools.github.io/managing-RSAT/genome_installation/install_organisms_FASTA_GTF.html).

This document **does not** use the following scripts, which are documented in
[Installing genomes and variations from Ensembl](https://rsa-tools.github.io/managing-RSAT/genome_installation/install_organisms_from_ensembl.html): 

- `install-ensembl-genome`, 
- `download-ensembl-genome`,
- `download-ensembl-features`, 
- `download-ensembl-variations`.


## Installing genome sequences and annotations from Ensembl Genomes

### Check FTP site URL

The current FTP site is at ftp://ftp.ensemblgenomes.org ; should it change in the future, or its folder structure, 
[ensemblgenomes_FTP_client.mk](https://github.com/rsa-tools/rsat-code/blob/master/makefiles/ensemblgenomes_FTP_client.mk)
will be updated accordingly.

### Find out current Ensembl Genomes release 

Visit the Web site of your division of interest, such as http://plants.ensembl.org, and check the release statement at the bottom.
Alternatively you can use the REST [info/eg_version endpoint](http://rest.ensembl.org/documentation/info/eg_version)

Ensembl Genome releases are integers and have an offset of 53 with Ensembl releases. 
For instance, Ensembl release 99 corresponds to Ensembl Genomes 46.

```{r, engine='bash', eval=FALSE}
export EGRELEASE=46
```

### Get supported species 

We shall download the current list of supported genomes in the respective division (GROUP).
This can be done by running the following command:

```{r, engine='bash', eval=FALSE}
cd $RSAT
export EGDIVISION=Plants

make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE organisms

# in the example, this will create $RSAT/data/ensemblgenomes/plants/release-46/species_EnsemblPlants.txt
```

Once this is done then you can install genomes from that release.

## Install all species 

You can install all genomes with these commands:

```{r, engine='bash', eval=FALSE}
cd $RSAT

# download FASTA and GTF files
nohup make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE download_all_species

# parse input files, extract genomic features and compute oligo frequencies
nohup make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE install_all_species

# check upstream sequences can be retrieved
nohup make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE check_all_species
```

The newly installed species will be added to $RSAT/data/supported_organisms.tab and should be listed with the following 
command-line:

```{r, engine='bash', eval=FALSE}
supported-organisms
```

**Note:** this will take a very long time, many days for a complete Ensembl Genomes release, so it might not be a good idea.

## Install selected species

In case you want to install a single genome you can do that by:

```{r, engine='bash', eval=FALSE}
cd $RSAT
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE SPECIES=oryza_longistaminata download_one_species
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE SPECIES=oryza_longistaminata install_one_species
```

You can also install selected genomes from older releases:

```{r, engine='bash', eval=FALSE}
cd $RSAT
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=42 organisms
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=42 SPECIES=oryza_sativa download_one_species
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=42 SPECIES=oryza_sativa install_one_species
```

As indicated earlier, the newly installed species are added to $RSAT/data/supported_organisms.tab 
and should appear in the list produced by:

```{r, engine='bash', eval=FALSE}
supported-organisms
```

## Other optional tasks

### Compute genome stats report

This step generates a report of descriptive stats of genomes currently in your system, 
such as http://rsat.eead.csic.es/plants/data/stats :
```{r, engine='bash', eval=FALSE}
make -f makefiles/ensemblgenomes_FTP_client.mk calc_stats
```

### Install variation data

If variation data is available for your species of interest you can download it with:

```{r, engine='bash', eval=FALSE}
cd $RSAT
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE SPECIES=oryza_sativa variations_one_species
```

**Note:** this will update file $RSAT/data/supported_organisms.tab

### Download and install Ensembl Compara homologies (optional)

Script `get-orthologs-compara` can be used to retrieve homologues (orthologues by default) precomputed at Ensembl Compara.
In order to use it you must first install Compara in your system, which you can do with:

```{r, engine='bash', eval=FALSE}
cd $RSAT
export ENSRELEASE=99

make -f makefiles/ensemblgenomes_FTP_client.mk RELEASE=$EGRELEASE ENSEMBL_RELEASE=$ENSRELEASE GROUP=$EGDIVISION download_compara
make -f makefiles/ensemblgenomes_FTP_client.mk RELEASE=$EGRELEASE ENSEMBL_RELEASE=$ENSRELEASE GROUP=$EGDIVISION parse_compara_match
make -f makefiles/ensemblgenomes_FTP_client.mk RELEASE=$EGRELEASE ENSEMBL_RELEASE=$ENSRELEASE GROUP=$EGDIVISION install_compara

```

All going well you can check the species with installed homologies with:

```{r, engine='bash', eval=FALSE}
get-orthologs-compara -supported_organisms
```

### Install Gene Ontology terms from BioMart

First we shall get the current registry for the division of interest. Only Plants, Metazoa and Fungi are supported.
For instance, for Plants the registry is at http://plants.ensembl.org/biomart/martservice?type=registry

The XML content of the registry must be copied and pasted into file *$RSAT/ext_lib/biomart-perl/conf/martURLLocation.xml* .
For instance, for Plants release 46 it will look like:
```
<?xml version="1.0" encoding="UTF-8"?>

<MartRegistry>
  <MartURLLocation database="plants_mart_46" default="" displayName="Ensembl Plants Genes 46" host="plants.ensembl.org" includeDatasets="" martUser="" name="plants_mart" path="/biomart/martservice" port="80" serverVirtualSchema="plants_mart" visible="1" />
  <MartURLLocation database="plants_snp_mart_46" default="" displayName="Ensembl Plants Variations 46" host="plants.ensembl.org" includeDatasets="" martUser="" name="plants_variations" path="/biomart/martservice" port="80" serverVirtualSchema="plants_mart" visible="1" />
  <MartURLLocation database="plants_sequence_mart_46" default="" displayName="Ensembl Plants Sequences 46" host="plants.ensembl.org" includeDatasets="" martUser="" name="plants_sequences" path="/biomart/martservice" port="80" serverVirtualSchema="plants_mart" visible="0" />
  <MartURLLocation database="plants_genomic_features_mart_46" default="" displayName="Ensembl Plants Genomic Features 46" host="plants.ensembl.org" includeDatasets="" martUser="" name="plants_genomic_features" path="/biomart/martservice" port="80" serverVirtualSchema="plants_mart" visible="0" />
  <MartURLLocation database="ontology_mart_99" default="" displayName="Ontology Mart 99" host="plants.ensembl.org" includeDatasets="" martUser="" name="ontology" path="/biomart/martservice" port="80" serverVirtualSchema="plants_mart" visible="0" />
</MartRegistry>
```

Once this is done we shall update the registry cache with:
```{r, engine='bash', eval=FALSE}
download-ensembl-go-annotations-biomart -reg
```
This will take a few minutes, but then we can do:
```{r, engine='bash', eval=FALSE}
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE download_go
```

And for each species for which we want GO terms we can now do:
```{r, engine='bash', eval=FALSE}
make -f makefiles/ensemblgenomes_FTP_client.mk GROUP=$EGDIVISION RELEASE=$EGRELEASE SPECIES=oryza_sativa download_go_annotations
```

*********
