<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2024-05-06" />

<title>Install organisms from NCBI</title>

<script src="install_organisms_from_ncbi_files/header-attrs-2.25/header-attrs.js"></script>
<script src="install_organisms_from_ncbi_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="install_organisms_from_ncbi_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="install_organisms_from_ncbi_files/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="install_organisms_from_ncbi_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="install_organisms_from_ncbi_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="install_organisms_from_ncbi_files/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">



<h1 class="title toc-ignore">Install organisms from NCBI</h1>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2024-05-06</h4>

</div>


<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The goal of this tutorial is to explain how to install genomes from
NCBI in your RSAT server.</p>
</div>
<div id="prerequisite" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Prerequisite</h1>
<p>This protocol assumes that you have a working instance of the RSAT
server, and that you are familar with the Unix terminal.</p>
</div>
<div id="protocol" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Protocol</h1>
<p>We illustrate the installation procedure with a bacterial genome,
since NCBI is the main source of prokaryote genomes for RSAT (<a
href="http://prokaryotes.rsat.eu/"
class="uri">http://prokaryotes.rsat.eu/</a>). The same procedure can
however apply to other taxa, as far as they are available on the NCBI
FTP server (<a href="https://ftp.ncbi.nlm.nih.gov/genomes/"
class="uri">https://ftp.ncbi.nlm.nih.gov/genomes/</a>).</p>
<div id="data-source" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Data source</h2>
<p>Genomes are downloaded from the refseq repository: <a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq"
class="uri">https://ftp.ncbi.nlm.nih.gov/genomes/refseq</a></p>
</div>
<div id="getting-the-list-of-available-species" class="section level2"
number="3.2">
<h2><span class="header-section-number">3.2</span> Getting the list of
available species</h2>
<p>The first step to perform with the command
<code>ìnstall-organisms</code> is to get the list of organisms available
for a given taxonomic group.</p>
<p>The NCBI FTP genome site is organised by “main” <strong>taxonomic
groups</strong>:</p>
<ul>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/">archaea</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/">bacteria</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/">fungi</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/invertebrate/">invertebrate</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/plant/">plant</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/mitochondrion/">mitochondrion</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/plasmid">plasmid</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/plastid">plastid</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/protozoa/">protozoa</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/">vertebrate_mammalian</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_other/">vertebrate_other</a></li>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/viral/">viral</a></li>
</ul>
<p>The option <code>-task available</code> returns the complete list of
available species for the specified group.</p>
<p>The command below gets the list of all available bacteria and stores
them in a text file. It can easily be adapted to another group by
changing the environment variable <code>GROUP</code>.</p>
<div class="sourceCode" id="cb1"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">## Get curret date to use as suffix for output files</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">TODAY</span><span class="op">=</span><span class="kw">`</span><span class="fu">date</span> +%Y-%m-%d<span class="kw">`</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">## Choose a taxonomic group of interest</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="bu">export</span> <span class="va">GROUP</span><span class="op">=</span>bacteria</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">## Get the list of bacterial species available on the NCBI FTP site</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="at">-task</span> available</span></code></pre></div>
<p>This will download the <strong>assembly summmary</strong> file from
NCBI, for the selected taxonomic group. Beware, for Bacteria this file
is pretty large (153Mb on April 2024). However, this file is downloaded
only if the version on the NCBI server is more recent than the local
version. After the first download, subsequent calls of the
<em>install-organisms</em> command will check if a new version is
present at the NCBI site, and if not it will use the local copy.</p>
<p>The output of <em>install-organisms</em> indicates the local path of
the assembly summary file.</p>
<div class="sourceCode" id="cb2"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="kw">;</span>    <span class="ex">Assembly</span> summary file      <span class="va">$RSAT</span>/downloads/refseq/bacteria/assembly_summary.txt</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="kw">;</span>    <span class="ex">Assembly</span> summary columns   <span class="va">$RSAT</span>/downloads/refseq/bacteria/assembly_summary_colums.tsv</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="kw">;</span>    <span class="ex">Available</span> assemblies         350727</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">## Count the number of species.</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">## Note: grep is used to filter out the comment lines.</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="bu">export</span> <span class="va">ASSEMBLY_SUMMARY</span><span class="op">=</span>downloads/refseq/<span class="va">${GROUP}</span>/assembly_summary.txt</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^#&#39;</span> <span class="va">${ASSEMBLY_SUMMARY}</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
<div id="getting-lists-of-available-species-for-other-groups"
class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Getting lists of
available species for other groups</h3>
<p>We can of course do the same for other taxa, for example Archaea,
Fungi or Mammalian.</p>
<div class="sourceCode" id="cb4"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">NCBI_GROUPS</span><span class="op">=</span><span class="st">&#39;archaea bacteria fungi invertebrate plant protozoa vertebrate_mammalian vertebrate_other viral&#39;</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;NCBI_GROUPS </span><span class="va">${NCBI_GROUPS}</span><span class="st">&quot;</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="cf">for</span> ncbi_group <span class="kw">in</span> <span class="va">$NCBI_GROUPS</span><span class="kw">;</span> <span class="cf">do</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">&quot;Getting list of NCBI species for </span><span class="va">${ncbi_group}</span><span class="st">&quot;</span> <span class="kw">;</span> <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> <span class="va">${ncbi_group}</span> <span class="at">-task</span> available</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">## Approx count of species per taxa</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">## (each file starts with 15 comment lines, which are be substracted from each count)</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="va">$RSAT</span>/downloads/refseq/<span class="pp">*</span>/assembly_summary.txt <span class="dt">\</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-nr</span> <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print &quot;|\t&quot;$1-15&quot;\t| &quot;$2&quot;\t|&quot;}&#39;</span> </span></code></pre></div>
<p>Note: for some species, several strains have been sequenced. The
number of genomes available at NCBI thus exceeds by far the number of
distinct species.</p>
</div>
<div id="number-of-species-per-group" class="section level3"
number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Number of species
per group</h3>
<p>Here is the result on April 24, 2024</p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead>
<tr class="header">
<th>Number of species</th>
<th>Taxonomic group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>350714</td>
<td>downloads/refseq/bacteria/assembly_summary.txt</td>
</tr>
<tr class="even">
<td>14979</td>
<td>downloads/refseq/viral/assembly_summary.txt</td>
</tr>
<tr class="odd">
<td>2016</td>
<td>downloads/refseq/archaea/assembly_summary.txt</td>
</tr>
<tr class="even">
<td>583</td>
<td>downloads/refseq/fungi/assembly_summary.txt</td>
</tr>
<tr class="odd">
<td>388</td>
<td>downloads/refseq/invertebrate/assembly_summary.txt</td>
</tr>
<tr class="even">
<td>385</td>
<td>downloads/refseq/vertebrate_other/assembly_summary.txt</td>
</tr>
<tr class="odd">
<td>216</td>
<td>downloads/refseq/vertebrate_mammalian/assembly_summary.txt</td>
</tr>
<tr class="even">
<td>169</td>
<td>downloads/refseq/plant/assembly_summary.txt</td>
</tr>
<tr class="odd">
<td>83</td>
<td>downloads/refseq/protozoa/assembly_summary.txt</td>
</tr>
<tr class="even">
<td><strong>369653</strong></td>
<td><strong>total</strong></td>
</tr>
</tbody>
</table>
<p><strong>Notes</strong></p>
<ul>
<li><p>The <strong>number of available assemblies</strong> exceeds by
far the number of species, because some species are represented by
different strains, in particular for bacteria (some bacterial species
have several thousands of strains sequenced).</p></li>
<li><p>Some of these genomes are only partly sequenced, or poorly
assembled, and the level of annotation is quite variable. NCBI
subdivided these available genomes in 4 categories depending on the
level of completion of the sequencing project.</p></li>
</ul>
</div>
</div>
<div id="filtering-assemblies" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Filtering
assemblies</h2>
<p>In order to select a reasonable number of genomes to install, we can
filter the assemblies based on different criteria.</p>
<div id="listing-the-available-assemblies-for-a-species-of-interest"
class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Listing the
available assemblies for a species of interest</h3>
<p>The option <code>-species</code> enables selecting a subset of
assemblies corresponding to a given species of a given genus. The
argulent <code>-species</code> should be followed by a string
concatenating the Genus (with a leading uppercase) and species (in
lowercase) separated by an underscore character (<code>_</code>), with
exactly the same spelling and cases as on the NCBI FTP refseq site. Note
that special characters (<code>][)(/</code> etc) should be replaced by
an underscore.</p>
<div class="sourceCode" id="cb5"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">## List the assemblies for a given species, e.g. Rhizobium etli</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">-species</span> Rhizobium_etli <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">-task</span> list <span class="at">-list_format</span> org</span></code></pre></div>
<p>Beware, for some species of interest, the number of assemblies can be
quite large. This is for example the case for <em>Escherchia
coli</em>.</p>
<div class="sourceCode" id="cb6"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">## List the assemblies for Escherichia coli)</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">-species</span> Escherichia_coli <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">-task</span> list <span class="at">-list_format</span> org <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">-o</span> assemblies_Escherichia_coli.txt</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">## Count the number of assemblies</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^;&#39;</span> assemblies_Escherichia_coli.txt <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^#&#39;</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">## Display the first rows of this file</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 20 assemblies_Escherichia_coli.txt</span></code></pre></div>
<p>For the species <em>Escherichia_coli</em>, we get 34510 assemblies
(05/05/2024).</p>
<p>We generally don’t need to install all these assemblies. We should
thus use additional information in order to narrow down the number of
genomes to install. To this purpose, the next sections explain how to
select genomes labeled either as <strong>reference</strong> or
<strong>representative</strong>, and completely sequenced.</p>
</div>
<div id="selecting-reference-genomes" class="section level3"
number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Selecting reference
genomes</h3>
<p>The assembly summary includes a field <code>refseq_category</code>,
which is quite convenient</p>
<div class="sourceCode" id="cb7"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria  <span class="at">-task</span> list <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="at">-list_format</span> org <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="at">-filter</span> refseq_category <span class="st">&quot;reference genome&quot;</span></span></code></pre></div>
<p>This will print out the list of assemblies labeled as “reference
genome” in the column “refseq_category” of the bacterial genome assembly
file.</p>
<p>As of 05/05/2024, no more than 15 bacterial genomes are labeled
“reference”, among the 350,729 available assemblies.</p>
<p>The option -list_format enables to export the result in 3 alternative
formats.</p>
<ul>
<li><p><em>org</em> a list of organism names (default output, as shown
above)</p></li>
<li><p><em>table</em> a table with the different names and identifiers
of the selected assemblies</p></li>
</ul>
<div class="sourceCode" id="cb8"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria  <span class="at">-task</span> list <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="at">-list_format</span> table <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">-filter</span>  refseq_category <span class="st">&quot;reference genome&quot;</span>  <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">-o</span> reference_genomes.tsv</span></code></pre></div>
<ul>
<li><em>bash</em> a bash script with the commands to install each of the
selected assemblies</li>
</ul>
<div class="sourceCode" id="cb9"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria  <span class="at">-task</span> list <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="at">-list_format</span> bash <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="at">-filter</span>  refseq_category <span class="st">&quot;reference genome&quot;</span> <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="at">-o</span> reference_genomes.bash</span></code></pre></div>
</div>
<div
id="listing-the-available-assemblies---representative-and-complete-genomes"
class="section level3" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Listing the
available assemblies - representative and complete genomes</h3>
<p>We can combine several filtering criteria to select a more reasonable
number of assemblies. For example, we could select the assemblies
labeled as “representative genome” in the column refseq_category, and
“Complete Genome” in the colum “”</p>
<div class="sourceCode" id="cb10"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria  <span class="at">-task</span> list <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="at">-list_format</span> table <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="at">-filter</span> refseq_category <span class="st">&quot;representative genome&quot;</span> <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="at">-filter</span> assembly_level <span class="st">&quot;Complete Genome&quot;</span> <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="at">-o</span> representative_complete.tsv</span></code></pre></div>
<p>This returns 5014 assemblies (24/04/2024), with only one assembly per
species, and belonging to 1555 distinct genus. The number of assemblies
per genus can be displayed with the following command;</p>
<div class="sourceCode" id="cb11"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^;&#39;</span> representative_complete.tsv <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^#&#39;</span> <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 6 <span class="kw">|</span> <span class="fu">sort</span>  <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-c</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-nr</span> <span class="kw">|</span> <span class="fu">less</span></span></code></pre></div>
</div>
</div>
<div
id="installing-one-assembly-for-a-given-species-and-strain-of-interest"
class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Installing one
assembly for a given species and strain of interest</h2>
<div id="downloading-a-selected-assembly" class="section level3"
number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Downloading a
selected assembly</h3>
<p>We will now download the genome of a selected assembly, and install
it on the RSAT server. As an illustration, we will install the reference
strain K-12 of <em>Escherichia coli</em>.</p>
<div class="sourceCode" id="cb12"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co">## Select a species and assembly of interest </span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">## (can be adapted to your needs)</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">grep</span> Escherichia reference_genomes.tsv <span class="kw">|</span> <span class="fu">grep</span> K-12</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="fu">grep</span> Escherichia reference_genomes.tsv <span class="kw">|</span> <span class="fu">grep</span> K-12 <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 8</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="bu">export</span> <span class="va">ASSEMBLY</span><span class="op">=</span>GCF_000005845.2</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">## List the selected assemblies for a given species </span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">## (e.g. Escherichia_coli)  </span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria <span class="dt">\</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>  <span class="at">-filter</span> assembly_accession <span class="va">${ASSEMBLY}</span> <span class="dt">\</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>  <span class="at">-task</span> list</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="co">## Download the selected assemblies </span></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> bacteria <span class="dt">\</span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a>  <span class="at">-filter</span> assembly_accession <span class="va">${ASSEMBLY}</span> <span class="dt">\</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a>  <span class="at">-task</span> download</span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a>  </span></code></pre></div>
<p><strong>Note</strong>: downloaded genomes will be installed in a
folder <code>$RSAT/downloads/refseq</code> on your computer. Each strain
is stored in a specific subfolder built from the group, genus, species,
and assembly.</p>
<p>For example, the reference assembly for <em>Escherichia coli</em> is
stored in</p>
<p><code>downloads/refseq/bacteria//Escherichia/Escherichia_coli_str._K-12_substr._MG1655/GCF_000005845.2_ASM584v2/</code></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>ls <span class="sc">-</span><span class="dv">1</span> downloads<span class="sc">/</span>refseq<span class="sc">/</span>bacteria<span class="sc">/</span>Escherichia<span class="sc">/</span>Escherichia_coli_str._K<span class="dv">-12</span>_substr._MG1655<span class="sc">/</span>GCF_000005845<span class="fl">.2</span>_ASM584v2</span></code></pre></div>
</div>
<div id="parsing-a-selected-genome-from-ncbi" class="section level3"
number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Parsing a selected
genome from NCBI</h3>
<p>Parsing consists in extracting the information from a text-formatted
file, in order to organise it in a computable structure.</p>
<div class="sourceCode" id="cb14"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co">## Parse a given strain of a given genome </span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="co">## (e.g. strain GCF_000016185.1_ASM1618v1 of Rhodospirillum centenum) </span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> bacteria <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">-filter</span> assembly_accession <span class="va">${ASSEMBLY}</span> <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="at">-task</span> parse</span></code></pre></div>
</div>
<div id="organism-names" class="section level3" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Organism names</h3>
<p>For each parsed genome, the program creates a folder whose name is
the concatenation of the genus, species, strain accession and assembly
name, separated by undescore characters.</p>
<p><code>$RSAT/public_html/data/[Genus]_[species]_[assembly_accession]_[asm_name]</code></p>
<p>For <em>Escherichia coli K12</em> reference strain, this gives:</p>
<p><code>Escherichia_coli_str._K-12_substr._MG1655_GCF_000005845.2_ASM584v2</code></p>
<p>From now on we will directly refer to this concatenated name as the
<strong><em>Organism name</em></strong> (generally denoted by the option
<code>-org</code> in RSAT commands.</p>
<div class="sourceCode" id="cb15"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">ORG</span><span class="op">=</span>Escherichia_coli_str._K-12_substr._MG1655_GCF_000005845.2_ASM584v2</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">## Check the parsing result</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> <span class="va">${RSAT}</span>/public_html/data/genomes/<span class="va">${ORG}</span>/genome/</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="co">## Measure the disk space occupied by this genome (note: this will be increased in the subsequent steps)</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a><span class="fu">du</span> <span class="at">-sm</span> <span class="va">${RSAT}</span>/public_html/data/genomes/<span class="va">${ORG}</span>/genome/</span></code></pre></div>
<p>This gives the number of megabases occupied by the genome folder.</p>
</div>
<div id="adding-the-parsed-genome-to-the-list-of-supported-genomes"
class="section level3" number="3.4.4">
<h3><span class="header-section-number">3.4.4</span> Adding the parsed
genome to the list of supported genomes</h3>
<p>In order to make the parsed genome available for RSAT, you need to
run the configuration task.</p>
<div class="sourceCode" id="cb16"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co">## Configure the organism for RSAT</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-org</span> <span class="va">${ORG}</span> <span class="at">-source</span> NCBI <span class="at">-task</span> config</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="co">## Check that the organism has wel been declared</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a><span class="ex">supported-organisms</span> <span class="at">-return</span> ID,source,last_update <span class="kw">|</span> <span class="fu">grep</span> <span class="va">${ORG}</span></span></code></pre></div>
</div>
<div id="finalizing-the-installation" class="section level3"
number="3.4.5">
<h3><span class="header-section-number">3.4.5</span> Finalizing the
installation</h3>
<p>Several more tasks need to be performed before we can consider this
organism as “installed”.</p>
<p>The simplest way to run it is to select the “default” tasks.</p>
<div class="sourceCode" id="cb17"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co">## Run the installation task (retrieve upstream sequences compute background oligo and dyad frequencies, index fasta genome for bedtols, ...)</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-org</span> <span class="va">${ORG}</span> <span class="at">-task</span> default</span></code></pre></div>
</div>
<div
id="detailed-list-of-supported-installation-tasks-default-and-others"
class="section level3" number="3.4.6">
<h3><span class="header-section-number">3.4.6</span> Detailed list of
supported installation tasks (default and others)</h3>
<table>
<colgroup>
<col width="20%" />
<col width="79%" />
</colgroup>
<thead>
<tr class="header">
<th>Task</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>default</td>
<td>run all the default tasks useful for the installation of a genome
(this is a subset of the tasks specified below)</td>
</tr>
<tr class="even">
<td>start_stop</td>
<td>compute trinucleotide frequencies in all the start and stop codons
(validation of the correspondance between genome sequences and gene
coordinates)</td>
</tr>
<tr class="odd">
<td>allup</td>
<td>Retrieve all upstream sequences in order to compute the background
models (oligonucleotide and dyad frequencies)</td>
</tr>
<tr class="even">
<td>seq_len_distrib</td>
<td>Compute the distribution of upstream sequences</td>
</tr>
<tr class="odd">
<td>genome_segments</td>
<td>SPlit the genome into segments corresponding to different genomic
region types: genic, intergenic, …</td>
</tr>
<tr class="even">
<td>upstream_freq</td>
<td>Compute oligonucleotide or/and dyad frequencies in upstream
sequences (must be combined with tasks oligos and/or dyads)</td>
</tr>
<tr class="odd">
<td>genome_freq</td>
<td>Compute oligonucleotide or/and dyad frequencies in the whole genome
(must be combined with tasks oligos and/or dyads)</td>
</tr>
<tr class="even">
<td>protein_freq</td>
<td>Compute amino acid, di- and tri-peptide frequencies in all protein
sequences</td>
</tr>
<tr class="odd">
<td>protein_len</td>
<td>Compute distribution of protein lengths</td>
</tr>
<tr class="even">
<td>oligos</td>
<td>Compute oligonucleotide frequencies (size 1 to 8) in a given type of
sequences. Must be combined with stasks upstream or genomic</td>
</tr>
<tr class="odd">
<td>dyads</td>
<td>Idem for dyads, i.e. spaced pairs of trinucleotides</td>
</tr>
<tr class="even">
<td>fasta_genome</td>
<td>convert genome sequence in fasta in order to make it usable by
bedtools</td>
</tr>
<tr class="odd">
<td>chrom_sizes</td>
<td>compute chromosome sizes from genomic sequences</td>
</tr>
<tr class="even">
<td>index_bedtools</td>
<td>index genome to enable its ultra-fast processing by bedtools</td>
</tr>
<tr class="odd">
<td>uninstall</td>
<td>uninstall a genome, i.e. stop displaying it as available. Beware,
this does not erase the genome folder, for this you can use the option
<code>-task erase</code></td>
</tr>
<tr class="even">
<td>erase</td>
<td>Erase the whole folder of your organism in
<code>$RSAT/public_html/data</code></td>
</tr>
</tbody>
</table>
</div>
<div id="checking-that-the-organism-is-well-supported"
class="section level3" number="3.4.7">
<h3><span class="header-section-number">3.4.7</span> Checking that the
organism is well supported</h3>
<div class="sourceCode" id="cb18"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co">## Get all the supported organisms</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="ex">supported-organisms</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="va">${ORG}</span></span></code></pre></div>
</div>
<div
id="checking-the-correctness-of-gene-annotations-by-computing-startstop-codon-frequencies"
class="section level3" number="3.4.8">
<h3><span class="header-section-number">3.4.8</span> Checking the
correctness of gene annotations by computing start/stop codon
frequencies</h3>
<div class="sourceCode" id="cb19"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co">## Retrieve all the start codon sequences</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="ex">retrieve-seq</span> <span class="at">-all</span> <span class="at">-org</span> <span class="va">${ORG}</span> <span class="at">-from</span> 0 <span class="at">-to</span> 2 <span class="at">-o</span> start_seq.fasta</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="co">## Check the result</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 20 start_seq.fasta</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="co">## Check that most genes have ATG as start codon</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="ex">oligo-analysis</span> <span class="at">-v</span> 1 <span class="at">-i</span> start_seq.fasta  <span class="at">-l</span> 3 <span class="at">-1str</span> <span class="at">-return</span> occ,freq <span class="at">-sort</span> <span class="kw">|</span> <span class="fu">more</span></span></code></pre></div>
<p>You can do the similar test with the stop codon frequencies. Note
that nothing forces you to store the stop codong sequences in a local
file, you can use the pipe symbol <code>|</code> to concatenate two
commands.</p>
<div class="sourceCode" id="cb20"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co">## Compute all trinucleotide frequencies in the stop codons</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="ex">retrieve-seq</span> <span class="at">-all</span> <span class="at">-org</span> <span class="va">${ORG}</span> <span class="dt">\</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>   <span class="at">-type</span> downstream <span class="at">-from</span> <span class="at">-1</span> <span class="at">-to</span> <span class="at">-3</span> <span class="at">-o</span> stop_seq.fasta</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="co">## Check the beginning of the result file</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a><span class="fu">head</span> <span class="at">-20</span> stop_seq.fasta</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a><span class="co">## Check that most genes have ATG as start codon</span></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="ex">oligo-analysis</span> <span class="at">-v</span> 1 <span class="at">-i</span> stop_seq.fasta  <span class="at">-l</span> 3 <span class="at">-1str</span> <span class="dt">\</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>   <span class="at">-return</span> occ,freq <span class="at">-sort</span> <span class="kw">|</span> <span class="fu">more</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="installing-multiple-genomes" class="section level2"
number="3.5">
<h2><span class="header-section-number">3.5</span> Installing multiple
genomes</h2>
<div id="selecting-reference-organisms-with-complete-genome"
class="section level3" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Selecting reference
organisms with complete genome</h3>
<p>The assembly summary file provided by NCBI for each taxonomic group
gives detailed indications of the sequencing status for each
assembly.</p>
<p>This table contains many columns with valuable information for genome
management. The column contain is indicated in a separate file.</p>
<ul>
<li><a
href="https://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt"
class="uri">https://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt</a></li>
</ul>
<p>Genome assemblies are classigied in four levels (<a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702866/">PMCID:
PMC4702866</a>).</p>
<ol style="list-style-type: decimal">
<li><strong>Contig:</strong> Assemblies that include only contigs.</li>
<li><strong>Scaffold:</strong> Includes both scaffolds and contigs.</li>
<li><strong>Chromosome:</strong> Includes chromosome or linkage groups,
plus scaffolds and contigs.</li>
<li><strong>Complete genome:</strong> Assemblies for which all molecules
are fully sequenced.</li>
</ol>
<p>We can use this assembly summary to select only those genomes with
the a completion level that we consider appropriate.</p>
<p>For example, for the RSAT prokaryote server (<a
href="http://prokaryote.rsat.eu"
class="uri">http://prokaryote.rsat.eu</a>), we restrict the installation
to genomes by combining the following criteria :</p>
<ul>
<li><p>The field <strong>refseq_category</strong> (column 5) is either
<em>reference genome</em> or <em>representative genome</em>.</p></li>
<li><p>The field <strong>assembly_level</strong> (column 12) is
<em>Complete genome</em></p></li>
</ul>
<p>Since ewe downloaded the assembly summary file above, we can count
the number of genomes of different types in these columns.</p>
<div class="sourceCode" id="cb21"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co">## Choose a taxonomic group of interest</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">GROUP</span><span class="op">=</span>bacteria</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="bu">export</span> <span class="va">ASSEMBLY_SUMMARY</span><span class="op">=</span>downloads/refseq/<span class="va">${GROUP}</span>/assembly_summary.txt</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="co">## Count the number of assemblies for each value of the refseq_category</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^#&#39;</span> <span class="va">${ASSEMBLY_SUMMARY}</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 5,12 <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span> <span class="fu">uniq</span> <span class="at">-c</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-n</span></span></code></pre></div>
<p>As of May 5, 2024, the number of assemblies in these categories is
the following:</p>
<div class="sourceCode" id="cb22"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>     <span class="ex">15</span> reference genome        Complete Genome</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>    <span class="ex">470</span> representative genome   Chromosome</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>   <span class="ex">4818</span> na                      Chromosome</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>   <span class="ex">5027</span> representative genome   Complete Genome</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>   <span class="ex">5195</span> representative genome   Scaffold</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>   <span class="ex">7940</span> representative genome   Contig</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>  <span class="ex">35394</span> na                      Complete Genome</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a> <span class="ex">105250</span> na                      Scaffold</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a> <span class="ex">186618</span> na                      Contig</span></code></pre></div>
</div>
<div id="installing-the-reference-genomes-with-complete-genome"
class="section level3" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Installing the
reference genomes with complete genome</h3>
<p>We saw above that we can apply two filters to select the 15 genomes
labeled as <em>reference genome</em> in the field
<code>refseq_category</code> and <em>Complete Genome</em> in
<code>assembly_level</code>.</p>
<p>We will first list the selected organisms in order to make sure that
the filter works properly (and avoid the risk of unwillingly installing
350,000 assemblies).</p>
<div class="sourceCode" id="cb23"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">GROUP</span><span class="op">=</span>bacteria</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="co">## Select the reference genomes with complete genome</span></span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>  <span class="at">-filter</span> refseq_category <span class="st">&quot;reference genome&quot;</span> <span class="dt">\</span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>  <span class="at">-filter</span> assembly_level <span class="st">&quot;Complete Genome&quot;</span> <span class="dt">\</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>  <span class="at">-task</span> list <span class="at">-list_format</span> org</span></code></pre></div>
<p>This should give a list of 15 genomes. We will then specity the tasks
required to install each of these genomes. Note that this can take some
time, since each genome has to be downloaded, parsed, and processed in
order to be ready for use on the RSAT server. In particular, the
computation of oligonucleotide and dyads in all the non-coding upstream
sequences can take some time.</p>
<div class="sourceCode" id="cb24"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co">## Install the reference genomes with complete genome</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 1 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="at">-filter</span> refseq_category <span class="st">&quot;reference genome&quot;</span> <span class="dt">\</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="at">-filter</span> assembly_level <span class="st">&quot;Complete Genome&quot;</span> <span class="dt">\</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>  <span class="at">-task</span> download,parse,config,default</span></code></pre></div>
<hr />
</div>
</div>
<div id="installing-many-genomes" class="section level2" number="3.6">
<h2><span class="header-section-number">3.6</span> Installing many
genomes</h2>
<p>We can install multiple species in a single command by specifying a
file containing a list of species names (one species name per row).</p>
<div id="download" class="section level3" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Download</h3>
<p><strong>Beware</strong>:</p>
<ul>
<li><p>This command will download thousands of genomes on your hard
drive, which will undoubtedly occupy some disj space.</p></li>
<li><p>The download can take a considerable time depending on the number
of selected species and bandwidth of your computer.</p></li>
</ul>
</div>
<div id="downloading-genome-files-from-ncbi" class="section level3"
number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Downloading genome
files from NCBI</h3>
<div class="sourceCode" id="cb25"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co">## File with the list of species to install</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="va">SPECIES_TO_INSTALL</span><span class="op">=</span><span class="va">${GROUP}</span>_species_to_install.txt</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a><span class="co">## Get the two first words of the 8th column as Genus and species names</span></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-f</span> 8 <span class="va">${REPRESENTATIVE_COMPLETE}</span> <span class="dt">\</span></span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/ /_/g&#39;</span> <span class="dt">\</span></span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>   <span class="op">&gt;</span> <span class="va">${SPECIES_TO_INSTALL}</span></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a><span class="co">## Count the number of species to instal</span></span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="va">${SPECIES_TO_INSTALL}</span></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a><span class="co">## Download all the selected species</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> download</span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>   </span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a><span class="co">## Check the list of downloaded organisms </span></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a><span class="co">## Note: organism = Genus + species + strain</span></span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> list</span></code></pre></div>
</div>
<div id="parsing" class="section level3" number="3.6.3">
<h3><span class="header-section-number">3.6.3</span> Parsing</h3>
<div class="sourceCode" id="cb26"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="co">## Parse all organisms</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> list,parse</span></code></pre></div>
</div>
<div id="parsing-genomes-from-ncbi-files" class="section level3"
number="3.6.4">
<h3><span class="header-section-number">3.6.4</span> Parsing genomes
from NCBI files</h3>
<p>Since the number of genomes to install can be important, we will run
<code>install-organism</code> with the option <code>-batch</code>, in
order to parallelise the installation steps via a job scheduler. This
job scheduler must have been previously defined in RSAT configuration
files.</p>
<div class="sourceCode" id="cb27"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> parse <span class="at">-batch</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="fu">make</span> <span class="at">-f</span> <span class="va">${RSAT}</span>/makefiles/server.mk  watch_jobs</span></code></pre></div>
</div>
<div id="default-installation-tasks" class="section level3"
number="3.6.5">
<h3><span class="header-section-number">3.6.5</span> Default
installation tasks</h3>
<div class="sourceCode" id="cb28"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="co">## Run default installation tasks</span></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co">## </span><span class="al">NOTE</span><span class="co">: if you need to install hundreds of genomes, this can take a while. ## In such case, it is worth testing the option -batch (see below)</span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> config,list,default</span></code></pre></div>
</div>
<div id="managing-installation-tasks-with-a-job-scheduler"
class="section level3" number="3.6.6">
<h3><span class="header-section-number">3.6.6</span> Managing
installation tasks with a job scheduler</h3>
<p>If your RSAT server has been configured adequately, you can add the
option <code>-batch</code> to the commands above in order to send the
installation tasks to a job scheduler.</p>
<div class="sourceCode" id="cb29"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="va">ORG_TO_INSTALL</span><span class="op">=</span><span class="va">${GROUP}</span>_organisms_to_install.txt</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a><span class="co">## Parse all organisms</span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>   <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> list,parse <span class="dt">\</span></span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>  <span class="at">-batch</span></span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a><span class="co">## Get the list of organisms from the list of parsed species</span></span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-group</span> <span class="va">${GROUP}</span> <span class="dt">\</span></span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>  <span class="at">-species_file</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="at">-task</span> list <span class="dt">\</span></span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>  <span class="op">&gt;</span> <span class="va">${ORG_TO_INSTALL}</span></span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a><span class="co">## Count the number of species and organisms to install</span></span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a><span class="va">SPECIES_DO_INSTALL_NB</span><span class="op">=</span><span class="kw">`</span><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^;&#39;</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span>  <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $1}&#39;</span><span class="kw">`</span></span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a><span class="va">ORG_DO_INSTALL_NB</span><span class="op">=</span><span class="kw">`</span><span class="fu">grep</span> <span class="at">-v</span> <span class="st">&#39;^;&#39;</span> <span class="va">${ORG_TO_INSTALL}</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $1}&#39;</span><span class="kw">`</span></span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;To install: </span><span class="va">${ORG_DO_INSTALL_NB}</span><span class="st"> organisms,  </span><span class="va">${SPECIES_DO_INSTALL_NB}</span><span class="st"> distinct species&quot;</span></span>
<span id="cb29-18"><a href="#cb29-18" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" tabindex="-1"></a><span class="co">## First retrieve all upstream sequences (cannot be done in batch mode)</span></span>
<span id="cb29-20"><a href="#cb29-20" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2  <span class="dt">\</span></span>
<span id="cb29-21"><a href="#cb29-21" tabindex="-1"></a>   <span class="at">-org_file</span> <span class="va">${ORG_TO_INSTALL}</span> <span class="at">-task</span> config,allup<span class="kw">;</span> </span>
<span id="cb29-22"><a href="#cb29-22" tabindex="-1"></a>   </span>
<span id="cb29-23"><a href="#cb29-23" tabindex="-1"></a><span class="co">## Send the other installation tasks to the job scheduler</span></span>
<span id="cb29-24"><a href="#cb29-24" tabindex="-1"></a><span class="ex">install-organism</span> <span class="at">-v</span> 2 <span class="at">-org_file</span> <span class="va">${ORG_TO_INSTALL}</span>  <span class="at">-task</span>  start_stop,seq_len_distrib,genome_segments,upstream_freq,genome_freq,protein_freq,protein_len,oligos,dyads,fasta_genome,fasta_genome_rm,chrom_sizes,index_bedtools <span class="at">-batch</span></span></code></pre></div>
</div>
</div>
<div id="check-the-installation" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> Check the
installation</h2>
<p>TO BE WRITTEN</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
